libs/wingsdk/src/std/array.ts

```diff
@@ -38,6 +38,19 @@ export class Array {
     throw new Error("Abstract");
   }
 
+  /**
+   * Try to get the value at the given index
+   *
+   * @macro ($self$[$args$])
+   *
+   * @param index index of the value to get
+   * @returns the value at the given index, or undefined if the index is out of bounds
+   */
+  public tryAt(index: number): T1 | undefined {
+    index;
+    throw new Error("Macro");
+  }
```

Comment:
Technically the current at() already does this, we just don't have it typed correctly. I guess we should add a macro to the current `at` to throw if undefined instead.

Reply:
This is included in the description for https://github.com/winglang/wing/issues/1868 which is already being worked on (not by me)